### PR TITLE
Small code cleanups & simplifications

### PR DESCRIPTION
**Pull request by MaxKellermann**

### CODE REVIEW

File: src/ClassicUO.Assets/MapLoader.cs [hunk 1]
```diff
@@ -415,7 +407,7 @@ public unsafe bool ApplyPatches(ref StackDataReader reader)
         {
             ResetPatchesInBlockTable();
 
-            PatchesCount = (int) reader.ReadUInt32BE();
+            int PatchesCount = (int) reader.ReadUInt32BE();
```
Comment (kamronbatman) [inline src/ClassicUO.Assets/MapLoader.cs]: Nit: Local variables should be camel cased.

Comment (MaxKellermann) [reply src/ClassicUO.Assets/MapLoader.cs]: OK, I'll change it. I wanted to keep the diff small, so I avoided using a different name.

File: src/ClassicUO.IO/DataReader.cs [hunk 2]
```diff
@@ -46,6 +46,8 @@ public unsafe class DataReader : IDisposable
         private byte* _data;
         private GCHandle _handle;
 
+        public bool HasData => _data != null;
```
Comment (kamronbatman) [inline src/ClassicUO.IO/DataReader.cs:49]: I don't think this check is correct. We should do an IntPtr.Zero check or use this:

https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.isnullref?view=net-8.0

**Review by kamronbatman — CHANGES_REQUESTED** [review]

File: src/ClassicUO.IO/DataReader.cs (hunk 2 above)
Comment (MaxKellermann) [reply src/ClassicUO.IO/DataReader.cs:49]: Isn't that the same, just a different (and more complicated) way to do it?

> This check is conceptually similar to (void*)(&source) == nullptr.

.... but what does "similar" mean?

Comment (MaxKellermann) [reply src/ClassicUO.IO/DataReader.cs:49]: note this compares the `byte*`, not the `IntPtr`.

Comment (MaxKellermann) [reply src/ClassicUO.IO/DataReader.cs:49]: I think the Microsoft documentation means that comparing `IntPtr` to `null` may implicitly convert the `IntPtr` to `IntPtr?` which is nullable; but this nulled value means the absence of an `IntPtr` value, which is different from `IntPtr.Zero` or just `0`.

But none of this applies to raw pointers. Comparing those to `null` is fine.

Comment (kamronbatman) [reply src/ClassicUO.IO/DataReader.cs:49]: Ah I see what you mean, the fact that it is a raw pointer, makes it valid to check against null vs a ref or a Intptr. Ok.

**Pull request merged**